### PR TITLE
Improve version check

### PR DESCRIPTION
**Pull request by hjoaquim**
This PR aims to: 

- improve the update check on the terminal by using the package `packaging.version`
- add unit tests to `terminal_helper.check_for_updates`

### CODE REVIEW

**Comment by hjoaquim** [comment]
awaits the merge of #3445

**Comment by hjoaquim** [comment]
many thanks @jmaslek

**Review by jmaslek — APPROVED** [review]
Nice

File: openbb_terminal/terminal_helper.py [hunk 1]
```diff
@@ -9,7 +9,8 @@
 import os
 import subprocess  # nosec
 import sys
-from typing import List, Optional
+from typing import List, Union
+from packaging import version
```
Comment (MichaMucha) [inline openbb_terminal/terminal_helper.py:13]: broken with `packaging==22` new version recently released

**Pull request merged**